eos-tech-support/eos-download-image

```diff
@@ -327,8 +352,16 @@ class EosDownloadImage(object):
         Prints the path to the downloaded OS image to stdout, to be consumed by
         eos-write-live-image.'''
 
-        _, images = self.fetch_manifest()
         args = self.args
+        if args.url is not None:
+            if args.size is None:
+                log("--size must be used when using --url")
+                sys.exit(1)
+            what = 'iso' if args.url.endswith('.iso') else 'full'
+            print(self.download_and_verify(meta=None, what=what, url=args.url, size=args.size))
```

Comment:
I think this block should be a separate function, `fetch_from_url(self)` and should call `download_and_verify` with a `meta` object it makes up.